File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> ans;
        set<int> myset;
        map<int,int> map1;
        map<int,int> map2;
        for(int i=0;i<nums1.size();i++){
            map1[nums1[i]]++;
            myset.insert(nums1[i]);
        }
        for(int i=0;i<nums2.size();i++){
            map2[nums2[i]]++;
        }
        for (const int &number : myset){
            int count=min(map1[number],map2[number]);
            if(count>0){
                while(count>0){
                    ans.push_back(number);
                    count--;
                }
            }
        }
        return ans;
    }
};

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.cs
public class Solution {
    public int[] Intersect(int[] nums1, int[] nums2) {
        Dictionary<int,int> d=new();
        foreach(var n in nums1){
            if(d.ContainsKey(n))
                d[n]++;
            else
                d.Add(n,1);
        }
        List<int> ans=new();
        foreach(var n in nums2){
            if(d.ContainsKey(n)){
                if(d[n]>0){
                    d[n]--;
                    ans.Add(n);
                }
            }
        }
        return ans.ToArray();
    }
}